stustapay/tse/tse_switchover.py

```diff
@@ -0,0 +1,263 @@
+                    print(f"{till['id']:5}:  {till['name']:50}")
+
+                print("ALL OF THESE TILLS NEED TO BE LOGGED OUT!")
+                input("confirm with enter when ready...")
+                # are you really sure?
+                print(f"Disabling TSE {self.tse_to_disable}, Serial {tse['tse_serial']}")
+                print("THIS CAN NOT BE UNDONE")
+                print("DO YOU REALY WANT TO PROCEED?")
+                confirmation = input("This is the last confirmation! Type 'yes' in uppercase: ")
+                if confirmation != "YES":
+                    print("Aborting...\n..\n.")
+                    return
+
+                # disable TSE
+                print("setting TSE to disabled")
+                if manualfailed == "YES":
+                    print("disabeling no matter what state the TSE has")
+                    await psql.execute("update tse set tse_status='disabled' where tse_name=$1", self.tse_to_disable)
+                else:
+                    print("disabeling only if in state failed")
+                    await psql.execute(
+                        "update tse set tse_status='disabled' where tse_name=$1 and tse_status='failed'",
+                        self.tse_to_disable,
+                    )
+
+                # check
+                tse = await psql.fetchrow("select * from tse where tse_name=$1", self.tse_to_disable)
+                if tse["tse_status"] == "disabled":
+                    print("SUCCESS: TSE set to disabled")
+                else:
+                    print("ERROR: could not set TSE to disabled, perhaps it was not in state failed")
+                    print("Aborting till reasignment")
+                    return
+
+                # reset till to tse assignment
+                print("resetting till to TSE assignment")
+                await psql.execute("update till set tse_id=NULL where tse_id=$1", tse["tse_id"])
+
+                # check
+                print("checking...")
+                tills = await psql.fetch("select id, name from till where tse_id=$1 order by id", tse["tse_id"])
+                if tills:
+                    print("there are still tills assigned to this TSE:")
+                    for till in tills:
+                        print(f"{till['id']:5}:  {till['name']:50}")
+                        print("Aborted")
+                        return
+                elif not tills:
+                    print("success, no more tills are assigned to this TSE")
+                else:
+                    raise RuntimeError("Error, this should not happen")
+
+                print("done")
+
+            if self.show:
+                tses = await psql.fetch("select * from tse")
+                print("TSE master data:")
+                for tse in tses:
+                    for key, item in tse.items():
+                        print(f"{key}: {item}")
+                        print("-------------------------------------------------------")
```

Comment:
i'd leave out this line, rather move it out of the loop.

Reply:
done